draft/phoenix_presence_with_live_view.md

```diff
@@ -0,0 +1,146 @@
+
+This is how we embed the LiveView Component into out markdown post:
+
+```elixir
+<%= Phoenix.LiveView.Helpers.live_render(@conn, RealWorldPhoenixWeb.Live.ReaderCount, session: %{slug: "simple_phoenix_presence"}) %>
+```
+
+## LiveView or LiveView Component?
+
+LiveView is maturing quickly and currently also has a concept of Components. Which basically are small building blocks that either are stateful or stateless. I am definitely planning on doing a writeup on using LiveView components, but for our usecase we simply need the basic LiveView.
+
+Let's create a basic LiveView boilerplate:
+
+```elixir
+defmodule RealWorldPhoenixWeb.Live.ReaderCount do
+  use Phoenix.LiveView
+
+  @moduledoc """
+    A small LiveView that shows the number of readers of a post using Phoenix Presence
+  """
+
+  def render(assigns) do
+    ~L"""
+      Readers: <\%= @reader_count %>
+    """
+  end
+
+  def mount(_session, socket) do
+    {:ok, assign(socket, :reader_count, 0)}
+  end
+end
+```
+
+Ok, so now we need to hook up Presence. Let's see how that would work. First we'll head over to the [docs](https://hexdocs.pm/phoenix/Phoenix.Presence.html) and see how it is setup. The first thing I notice is this: _Provides Presence tracking to processes and channels._. We are not really using Channels, but our LiveView is a process so I guess that should work, right?
+
+## Presence Setup
+
+Let's first setup Presence using the guide provides by the Phoenix Team:
+
+### 1. Setup a Presence Module in our app:
+
+```elixir
+defmodule RealWorldPhoenix.Presence do
+  use Phoenix.Presence, otp_app: :real_world_phoenix,
+                        pubsub_server: MyApp.PubSub
+end
+```
+
+### 2.  Add this module to our supervision tree:
+
+```elixir
+# lib/real_world_phoenix/application.ex
+
+defmodule RealWorldPhoenix.Application do
+  @moduledoc false
+  use Application
+
+  def start(_type, _args) do
+    children = [
+      RealWorldPhoenix.Repo,
+      RealWorldPhoenixWeb.Endpoint,
+      RealWorldPhoenix.Presence # <= Add This!
+    ]
+
+    opts = [strategy: :one_for_one, name: RealWorldPhoenix.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+
+  def config_change(changed, _new, removed) do
+    RealWorldPhoenixWeb.Endpoint.config_change(changed, removed)
+    :ok
+  end
+end
+```
+
+## I don't have any Channels... ?
+
+So the next and last step in the guide shows an example of adding Presence tracking to a channel implementation. But we are not using channels! So how do we go about and add this to our LiveView. Presence uses a PubSub (Publish-Subscribe) mechanism to perform it's magic. So what we should do in our LiveView is make sure we subscribe to the topic of the current page. Once we do that we can add a callback that will track any presence_diff where we will update the counter if someone leaves or joins the current topic.
```

Comment:
probably `perform it's magic.` should end with a `?` and the part after should be a new paragraph